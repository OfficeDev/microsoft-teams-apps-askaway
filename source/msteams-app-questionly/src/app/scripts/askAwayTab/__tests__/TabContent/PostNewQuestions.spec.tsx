/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { Flex, Text, Card, ThemePrepared } from '@fluentui/react-northstar';
import Adapter from 'enzyme-adapter-react-16';
import { PostNewQuestions } from '../../TabContent/PostNewQuestions';
import { Helper } from '../../shared/Helper';
import Badge from '../../shared/Badge';

configure({ adapter: new Adapter() });

describe('PostNewQuestions Component', () => {
    const activeSessionData = new Helper().createEmptyActiveSessionData();
    const onPostNewQuestion = jest.fn();
    const t = jest.fn();
    const theme = {} as ThemePrepared;

    it('should match the snapshot', () => {
        const wrapper = shallow(<PostNewQuestions t={t} theme={theme} activeSessionData={activeSessionData} onPostNewQuestion={onPostNewQuestion} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(<PostNewQuestions t={t} theme={theme} activeSessionData={activeSessionData} onPostNewQuestion={onPostNewQuestion} />);

        expect(component.find(Flex)).toHaveLength(3);
        expect(component.find(Text)).toHaveLength(2);
        expect(component.find(Card)).toHaveLength(1);
        expect(component.find(Badge)).toHaveLength(1);
    });
});
