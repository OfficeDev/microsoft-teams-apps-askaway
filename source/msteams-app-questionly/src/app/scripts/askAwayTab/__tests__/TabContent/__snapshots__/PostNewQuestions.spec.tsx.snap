// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostNewQuestions Component should match the snapshot 1`] = `
<div
  className="post-new-question"
>
  <Card
    accessibility={[Function]}
    aria-roledescription="card"
    className="card-layout"
    size="medium"
    style={
      Object {
        "backgroundColor": undefined,
        "borderColor": "",
      }
    }
  >
    <CardHeader
      fitted={true}
    >
      <Flex
        gap="gap.small"
      >
        <Flex
          column={true}
        >
          <Badge
            className="badge"
            styles={
              Object {
                "backgroundColor": undefined,
                "color": undefined,
              }
            }
          />
          <Text
            as="span"
            className="date-content-format"
            size="small"
          />
        </Flex>
      </Flex>
    </CardHeader>
    <CardBody>
      <Flex
        column={true}
        gap="gap.small"
      >
        <Text
          as="span"
          className="session-title"
          content=""
          size="large"
          weight="bold"
        />
      </Flex>
    </CardBody>
    <CardFooter
      styles={
        Object {
          "marginBottom": "0",
        }
      }
    >
      <Text
        as="span"
        content=""
      />
    </CardFooter>
  </Card>
</div>
`;
